app/policies/comment_policy.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class CommentPolicy < ApplicationPolicy
+  def update?
+    return record.approved_by_admin? if user.admin
```

Comment:
The readability on these can quickly deteriorate, I'd focus on keeping it clear and explicit.

```
if user.admin?
  return true if record.approved_by_admin?
else
  return true if record.approved_by_admin? && !record.not_proceeding?
end
```

This way we can swap the `if` out for a `case` statement on the `role` if new roles ever get added.

It would be nice to replace the `!record.not_proceeding?` with one that is the opposite, like `record.proceeding?` as well, just to keep it easy to understand.

Reply:
I will have a go, I seem to remember I had a similar statement before and Rubocop didn't like it, I'll try again and let you know if I hit any issues

Reply:
Ah, it might. Just leave it if it does.